src/ert/_c_wrappers/analysis/analysis_module.py

```diff
@@ -1,162 +1,216 @@
-        return items
-
-    def getVariableValue(self, name: "VariableName") -> Union[int, float, bool]:
-        self.__assertVar(name)
-        variable_type = self.getVariableType(name)
-        if variable_type == float:
-            return self.getDouble(name)
-        elif variable_type == bool:
-            return self.getBool(name)
-        elif variable_type == int:
-            return self.getInt(name)
+    if module == AnalysisMode.ENSEMBLE_SMOOTHER:
+        return es_variables
+    return ies_variables
+
+
+class AnalysisModule:
+    DEPRECATED_KEYS = ["USE_EE", "USE_GE"]
+    TRUNK_ALTERNATE_KEYS = ["ENKF_NCOMP", "ENKF_SUBSPACE_DIMENSION"]
+    SPECIAL_KEYS = ["INVERSION", *TRUNK_ALTERNATE_KEYS, *DEPRECATED_KEYS]
+
+    def __init__(self, mode: AnalysisMode, name: str, variables: Dict, iterable: bool):
+        self.mode = mode
+        self.name = name
+        self.iterable = iterable
+        self._variables = variables
+
+    @classmethod
+    def ens_smother_module(cls, name: AnalysisMode = AnalysisMode.ENSEMBLE_SMOOTHER):
+        return cls(
+            mode=AnalysisMode.ENSEMBLE_SMOOTHER,
+            name=name,
+            variables=get_mode_variables(AnalysisMode.ENSEMBLE_SMOOTHER),
+            iterable=False,
+        )
+
+    @classmethod
+    def iterated_ens_smother_module(
+        cls, name: AnalysisMode = AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER
+    ):
+        return cls(
+            mode=AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER,
+            name=name,
+            variables=get_mode_variables(AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER),
+            iterable=True,
+        )
+
+    def get_variable_names(self) -> List[str]:
+        return list(self._variables.keys())
+
+    def get_variable_value(self, name) -> Union[int, float, bool]:
+        if name in self._variables:
+            return self._variables[name]["value"]
+        raise KeyError(f"Variable {name} not found in module")
+
+    def variable_value_dict(self) -> Dict[str, Union[float, int]]:
+        return {name: var["value"] for name, var in self._variables.items()}
+
+    def handle_special_key_set(self, var_name, value):
+        if var_name in self.DEPRECATED_KEYS:
+            logger.warning(
+                "USE_EE/USE_GE settings have been removed"
+                "use the INVERSION setting instead"
+            )
+        elif var_name == "INVERSION":
+            inversion_str_map = {
+                "EXACT": 0,
+                "SUBSPACE_EXACT_R": 1,
+                "SUBSPACE_EE_R": 2,
+                "SUBSPACE_RE": 3,
+            }
+            if value in inversion_str_map:
+                self.set_var("IES_INVERSION", inversion_str_map[value])
+        elif var_name in self.TRUNK_ALTERNATE_KEYS:
+            self.set_var("ENKF_TRUNCATION", value)
+
+    def set_var(self, var_name: str, value: Union[float, int, bool, str]):
+        if var_name in self.SPECIAL_KEYS:
+            self.handle_special_key_set(var_name, value)
```

Comment:
The min/max is farther down in this function and it happens for all keys in `self._variables`

Reply:
right, exactly - and it doesn't happen to special keys though, since it's an `elif`, correct? is that OK? we don't need the checks there? because `INVERSION`, for instance, which is one of the special keys, has a min and max defined, so we could do the check?